### PR TITLE
Feature/map offset operator overloads

### PR DESCRIPTION
**Pull request by oscar139**
Added project file for tests for OPHD. And added the operator overloads for MapOffset for neater scalar multiplication.

### CODE REVIEW

File: OPHD/Map/MapOffset.h [hunk 1]
```diff
@@ -8,3 +8,36 @@ struct MapOffset
 	NAS2D::Vector<int> xy;
 	int z;
 };
+
```
Comment (ldicker83) [inline OPHD/Map/MapOffset.h:11]: Style guidelines: double spaces between methods and functions. Otherwise I'm not seeing anything blocking here. Would like to see some feedback from @DanRStevens before merging.

Comment (oscar139) [reply OPHD/Map/MapOffset.h:11]: Ok will fix. I add two more commits to actually use the new operators. Oh and we some how missed updating calls to the new MapViewState:moveView from mMapView::moveView in MapViewState::onKeyDown

File: OPHD/Map/MapOffset.h [hunk 2]
```diff
@@ -8,3 +8,41 @@ struct MapOffset
...
+}
+
+
+inline MapOffset operator*(const MapOffset& offset, int scalar)
+{
+	MapOffset result;
+	result.xy = offset.xy * scalar;
+	result.z = offset.z * scalar;
+	return result;
+}
```
Comment (DanRStevens) [inline OPHD/Map/MapOffset.h:31]: When you also support `operator*=`, you can use that to implement `operator*`. Simply remove the `const &` from the parameter passing so it generates a copy, and then use `operator*=` on the copy.

```cpp
inline MapOffset operator*(MapOffset offset, int scalar)
{
	return offset *= scalar;
}
```

File: OPHD/States/MapViewState.cpp [hunk 3]
```diff
@@ -421,22 +421,22 @@ void MapViewState::onKeyDown(NAS2D::EventHandler::KeyCode key, NAS2D::EventHandl
...
 		case NAS2D::EventHandler::KeyCode::KEY_a:
 		case NAS2D::EventHandler::KeyCode::KEY_LEFT:
-			mMapView->moveView(MapOffset{DirectionSouthWest * shiftMoveScalar, 0});
+			mMapView->moveView(MapOffset{DirectionSouthWest * shiftMoveScalar});
 			break;
 
 		case NAS2D::EventHandler::KeyCode::KEY_d:
 		case NAS2D::EventHandler::KeyCode::KEY_RIGHT:
-			mMapView->moveView(MapOffset{DirectionNorthEast * shiftMoveScalar, 0});
+			mMapView->moveView(MapOffset{DirectionNorthEast * shiftMoveScalar});
```
Comment (DanRStevens) [inline OPHD/States/MapViewState.cpp]: This doesn't appear to be using the new `operator*`. It's still using the existing `operator*` on the `NAS2D::Vector<int>` class. The only change is there is no explicit initialization value for `z`. Since the `struct` is initialized with `{}`, unspecified fields should be default initialized, which in this case should result in a `0` for `z`.

I assume what you meant to write was:
```cpp
mMapView->moveView(MapOffset{DirectionNorthEast, 0} * shiftMoveScalar);
```

Of course with the new `operator*` that means we can go back to using the predefined `MapOffset` values from `DirectionOffset.h`:
```cpp
mMapView->moveView(MapOffsetNorthEast * shiftMoveScalar);
```

File: OPHD/States/MapViewState.cpp [hunk 4]
```diff
@@ -741,7 +741,7 @@ void MapViewState::onChangeDepth(int oldDepth, int newDepth) {
 
 void MapViewState::moveView(MapOffset offset) {
 	int oldZLevel = mMapView->currentDepth();
-	
+
```
Comment (DanRStevens) [inline OPHD/States/MapViewState.cpp:744]: I approve of whitespace fixes. Preferably they should be in a separate commit from code changes.

File: testOPHD/testOPHD.cpp [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#include <gtest/gtest.h>
+
+
+TEST(Test, EmptyTest)
+{
+}
```
Comment (DanRStevens) [inline testOPHD/testOPHD.cpp:6]: Slightly awkward situation, as we still haven't completed:
- Issue #1422

Ideally, `MapOffset` would be migrated to `libOphd`, and so unit test code for it would go into `testLibOphd`. Unfortunately that hasn't happened yet.

----

In C++ it's a bit more natural to write unit tests against library code. I wouldn't expect there to be a `testOphd` project, since the `Ophd` project is an app rather than a library.

----

The `makefile` doesn't support building the new project. It does support building `testLibOphd`.

File: testOPHD/MapOffsetTests.cpp [hunk 6]
```diff
@@ -0,0 +1,105 @@
+#include "OPHD/Map/MapOffset.h"
```
Comment (DanRStevens) [inline testOPHD/MapOffsetTests.cpp:1]: As `MapOffset` is expected to be in library code, and the unit tests for that library are in a separate project, it would be more appropriate for the `#include` to use `<>` rather than `""`.

File: testOPHD/MapOffsetTests.cpp [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+{
+	MapOffset offset1;
+	offset1.xy = { 1, 2 };
+	offset1.z = 3;
+
+	MapOffset offset2;
+	offset2.xy = { 1, 2 };
+	offset2.z = 3;
+
+	EXPECT_TRUE(offset1 == offset2);
```
Comment (DanRStevens) [inline testOPHD/MapOffsetTests.cpp:16]: This would be more terse with inline construction:
```cpp
EXPECT_TRUE(MapOffset{{1, 2}, 3} == MapOffset{{1, 2}, 3});
```

----

Additionally, this particular check might use `EXPECT_EQ` instead of `EXPECT_TRUE`. Not that it's much shorter or really changes readability.

```cpp
EXPECT_EQ(MapOffset{{1, 2}, 3}, MapOffset{{1, 2}, 3});
```

----

Similar for other tests too. Use inline construction. Using more specific expectations, such as `EXPECT_NE` (not equal).

File: testOPHD/MapOffsetTests.cpp [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+	EXPECT_EQ(result3, expectedResult3);
+
+	int zeroScalar = 0;
+	MapOffset result4 = offset * zeroScalar;
+	MapOffset expectedResult4 = {0, 0, 0};
+	EXPECT_EQ(result4, expectedResult4);
+
+	MapOffset result5 = offset *= scalar;
+	MapOffset expectedResult5 = {-2, 4, 6};
+	EXPECT_EQ(result5, expectedResult5);
```
Comment (DanRStevens) [inline testOPHD/MapOffsetTests.cpp:104]: Might be wise to split tests for `operator*=` from the tests for `operator*`.
